From 83459d5d6092e614b5e1abe6394d24527adf8c9b Mon Sep 17 00:00:00 2001
From: Devi Priya <quic_devipriy@quicinc.com>
Date: Fri, 25 Aug 2023 18:11:25 +0530
Subject: [PATCH] drivers: soc: qcom: Add support for IPQ_DEBUG config

Add support for CONFIG_IPQ_DEBUG to enable/disable the
generic debug driver for IPQ targets.

Change-Id: I9ff587aa129381b36a5331df02368685ad65c5cf
Signed-off-by: Devi Priya <quic_devipriy@quicinc.com>
---
 drivers/soc/qcom/Kconfig  | 6 ++++++
 drivers/soc/qcom/Makefile | 1 +
 2 files changed, 7 insertions(+)

diff --git a/drivers/soc/qcom/Kconfig b/drivers/soc/qcom/Kconfig
index dd54e38b0963..0a03377c6a21 100644
--- a/drivers/soc/qcom/Kconfig
+++ b/drivers/soc/qcom/Kconfig
@@ -261,6 +261,12 @@ config IPQ_REG_UPDATE
 	help
 	 Generic driver to configure required registers
 
+config IPQ_DEBUG
+	bool "QTI, Subsystem IPQ debug driver"
+	depends on ARCH_QCOM
+	help
+	 Generic driver to support debug features
+
 config MSM_RPM_LOG
 	tristate "QTI Resource Power Manager Log Driver"
 	help
diff --git a/drivers/soc/qcom/Makefile b/drivers/soc/qcom/Makefile
index e75fce6d3374..ca974c66a751 100644
--- a/drivers/soc/qcom/Makefile
+++ b/drivers/soc/qcom/Makefile
@@ -31,5 +31,6 @@ obj-$(CONFIG_QCOM_KRYO_L2_ACCESSORS) +=	kryo-l2-accessors.o
 obj-$(CONFIG_QCOM_ICC_BWMON)	+= icc-bwmon.o
 obj-$(CONFIG_QCOM_DCC_V2)	+= dcc_v2.o
 obj-$(CONFIG_IPQ_REG_UPDATE) += ipq_reg_update.o
+obj-$(CONFIG_IPQ_DEBUG)	+= ipq-debug.o
 obj-$(CONFIG_MSM_RPM_LOG)	+= rpm_log.o
 obj-$(CONFIG_QTI_TZ_LOG) += tz_log.o
-- 
2.34.1

